src/wallet/wallet2.cpp

```diff
@@ -14942,56 +14942,155 @@ std::string wallet2::decrypt_with_view_secret_key(const std::string &ciphertext,
   return decrypt(ciphertext, get_account().get_keys().m_view_secret_key, authenticated);
 }
 //----------------------------------------------------------------------------------------------------
-std::string wallet2::make_uri(const std::string &address, const std::string &payment_id, uint64_t amount, const std::string &tx_description, const std::string &recipient_name, std::string &error) const
+std::string wallet2::custom_convert_to_url_format(const std::string &uri) const
 {
-  cryptonote::address_parse_info info;
-  if(!get_account_address_from_str(info, nettype(), address))
+  std::string s = epee::net_utils::conver_to_url_format(uri);
+
+  // replace '=' with "%3D" and '?' with "%3F".
+  std::string result;
+  result.reserve(s.size());
+
+  for (char c : s)
   {
-    error = std::string("wrong address: ") + address;
-    return std::string();
+    if (c == '=')
+    {
+      result.append("%3D");
+    }
+    else if (c == '?')
+    {
+      result.append("%3F");
+      }
+      
+    else
+    {
+      result.push_back(c);
+    }      
   }
 
-  // we want only one payment id
-  if (info.has_payment_id && !payment_id.empty())
+  return result;
+}
+//----------------------------------------------------------------------------------------------------
+std::string wallet2::make_uri(const std::vector<std::string> &addresses, const std::vector<std::uint64_t> &amounts, const std::vector<std::string> &recipient_names, const std::string &tx_description, std::string &error) const
+{
+  if (addresses.empty())
   {
-    error = "A single payment id is allowed";
+    error = "No recipient information like addresses are provided.";
     return std::string();
   }
 
-  if (!payment_id.empty())
+  if (addresses.size() != amounts.size() || addresses.size() != recipient_names.size())
   {
-    error = "Standalone payment id deprecated, use integrated address instead";
+    error = (boost::format("The counts of addresses (%zu), amounts (%zu), and recipient names (%zu) do not match.") % addresses.size() % amounts.size() % recipient_names.size()).str();
```

Comment:
```suggestion
    error = (boost::format("The counts of addresses (%1%), amounts (%2%), and recipient names (%3%) do not match.") % addresses.size() % amounts.size() % recipient_names.size()).str();
```

The current syntax appears incorrect. Please also update the other places that use `boost::format`.

https://www.boost.org/doc/libs/1_88_0/libs/format/doc/format.html